magellan-library/src/test/java/com/wealthfront/magellan/navigation/LinearNavigatorTest.kt

```diff
@@ -9,43 +11,71 @@ import com.wealthfront.magellan.ScreenContainer
 import com.wealthfront.magellan.core.Journey
 import com.wealthfront.magellan.core.Step
 import com.wealthfront.magellan.databinding.MagellanDummyLayoutBinding
+import com.wealthfront.magellan.view.ActionBarConfig
+import com.wealthfront.magellan.view.ActionBarModifier
 import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.anyString
 import org.mockito.Mock
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
 import org.mockito.MockitoAnnotations.initMocks
+import org.robolectric.Robolectric.buildActivity
 import org.robolectric.RobolectricTestRunner
+import org.robolectric.android.controller.ActivityController
+import org.robolectric.fakes.RoboMenu
 
 @RunWith(RobolectricTestRunner::class)
 class LinearNavigatorTest {
 
+  private val fakeId = 1000
+
   @Mock lateinit var screenContainer: ScreenContainer
 
+  private lateinit var activityController: ActivityController<FakeActivity>
+  private lateinit var menu: Menu
   private lateinit var step1: Step<*>
   private lateinit var step2: Step<*>
   private lateinit var journey1: Journey<*>
   private lateinit var step3: Step<*>
   private lateinit var journey2: Journey<*>
   private lateinit var step4: Step<*>
   private lateinit var linearNavigator: LinearNavigator
+  private lateinit var context: Activity
 
   @Before
   fun setUp() {
     initMocks(this)
-    linearNavigator = LinearNavigator { screenContainer }
-
-    step1 = DummyStep()
+    step1 = spy(DummyStep())
+    journey1 = spy(DummyJourney())
     step2 = DummyStep()
-    journey1 = DummyJourney()
     step3 = DummyStep()
     journey2 = DummyJourney()
     step4 = DummyStep()
+    activityController = buildActivity(FakeActivity::class.java)
+    context = spy(activityController.get())
+    menu = RoboMenu(context)
+    menu.add(0, fakeId, 0, "some string")
+
+    linearNavigator = LinearNavigator(journey1) { screenContainer }
+    linearNavigator.create(context)
   }
 
   @Test
   fun goTo() {
+    linearNavigator.menu = menu
+
+    verify(journey1 as ActionBarModifier).onUpdateMenu(menu)
+
     linearNavigator.goTo(step1)
 
+    verify(step1).setTitle(anyString())
+    verify(journey1 as ActionBarModifier, times(2)).onUpdateMenu(menu)
+    verify(step1 as ActionBarModifier).onUpdateMenu(menu)
+    verify(context as ActionBarConfigListener).onNavigate(any())
```

Comment:
Nit, not blocking: could we test this by having the journey/step modify the menu, rather than using spies?